### ISSUE TITLE
+21323290096 is reported to be invalid by the library

### ISSUE THREAD

**Issue by AlaaT** [googlei18n/libphonenumber#754]
Hello the Number **+21323290096** is reported as invalid number when its valid fixed_line number in Algeria .

**Code :** 

```xml
<!-- Algeria -->

<territory id="DZ" countryCode="213" internationalPrefix="00" nationalPrefix="0" nationalPrefixFormattingRule="$NP$FG">

   <references>

      <sourceUrl>http://www.itu.int/oth/T0202000003/en</sourceUrl>

      <sourceUrl>http://www.arpt.dz</sourceUrl>

   </references>

   <availableFormats>

      <!-- Formatting from http://www.pagesjaunes-dz.com. -->

      <numberFormat pattern="([1-4]\d)(\d{2})(\d{2})(\d{2})">

         <leadingDigits>[1-4]</leadingDigits>

         <format>$1 $2 $3 $4</format>

      </numberFormat>

      <numberFormat pattern="([5-8]\d{2})(\d{2})(\d{2})(\d{2})">

         <leadingDigits>[5-8]</leadingDigits>

         <format>$1 $2 $3 $4</format>

      </numberFormat>

      <numberFormat pattern="(9\d)(\d{3})(\d{2})(\d{2})">

         <leadingDigits>9</leadingDigits>

         <format>$1 $2 $3 $4</format>

      </numberFormat>

   </availableFormats>

   <generalDesc>

      <nationalNumberPattern>(?:

            [1-4]|

            [5-9]\d

          )\d{7}</nationalNumberPattern>

      <possibleNumberPattern>\d{8,9}</possibleNumberPattern>

   </generalDesc>

   <fixedLine>

      <!-- We include the VSAT lines here. -->

      <nationalNumberPattern>(?:

            1\d|

            2[014-79]|

            3[0-8]|

            4[0135689]

          )\d{6}|

          9619\d{5}</nationalNumberPattern>

      <exampleNumber>12345678</exampleNumber>

   </fixedLine>

   <mobile>

      <!-- Adding 65 and 78 from numbers found online. Also, prefix 670 is added since the carrier

             Mobilis Algeria provided it, and 54 for Nedjma. Prefixes 67[1-4] were also added for

             Mobilis Algeria based on evidence found online. -->

      <nationalNumberPattern>(?:

            5[4-6]|

            7[7-9]

          )\d{7}|

          6(?:

            [569]\d|

            7[0-4]

          )\d{6}</nationalNumberPattern>

      <possibleNumberPattern>\d{9}</possibleNumberPattern>

      <exampleNumber>551234567</exampleNumber>

   </mobile>

   <tollFree>

      <nationalNumberPattern>800\d{6}</nationalNumberPattern>

      <possibleNumberPattern>\d{9}</possibleNumberPattern>

      <exampleNumber>800123456</exampleNumber>

   </tollFree>

   <premiumRate>

      <nationalNumberPattern>80[3-689]1\d{5}</nationalNumberPattern>

      <possibleNumberPattern>\d{9}</possibleNumberPattern>

      <exampleNumber>808123456</exampleNumber>

   </premiumRate>

   <!-- The Algerian plan doesn't specify where the costs start to be considered "premium", so we

           draw an arbitrary line here and say that from 50 Da up they will be considered premium.

           -->

   <sharedCost>

      <nationalNumberPattern>80[12]1\d{5}</nationalNumberPattern>

      <possibleNumberPattern>\d{9}</possibleNumberPattern>

      <exampleNumber>801123456</exampleNumber>

   </sharedCost>

   <voip>

      <nationalNumberPattern>98[23]\d{6}</nationalNumberPattern>

      <possibleNumberPattern>\d{9}</possibleNumberPattern>

      <exampleNumber>983123456</exampleNumber>

   </voip>

</territory>
```

**Comment by keghani** [comment]
Hi,

Do you have supporting evidence of the number being valid? For example official documentation from the government, telecom authority or carrier are good. If none of them are available, news reports of the introduction of the new prefixes are also OK.

**Comment by NeverMindNever** [comment]
Hi ,

Thank you for your quick reply .

I found this document about numbering plan on algeria : 

http://www.arpt.dz/fr/doc/gd/num/New_plan_num_EN.pdf

and you can call this phone number to make sure i did it my self and the call was connected .

Best regards, 
Alaa

**Comment by keghani** [comment]
We consider valid the numbers beginning with 2 that are followed by 0, 1, 4-7, or 9.

The document you link to does not have details on valid phone number ranges. All it says about numbers beginning with 2 is that they may be of the form +213 2 BPQMCDU, but the "BPQMCDU" part could still have restrictions. Please find specific evidence for the range you want supported.

**Comment by keghani** [comment]
Hi again,

Just to check in on this, will you provide evidence for the range you want supported? Otherwise we can't make the required change and will have to close this issue.

**Comment by AlaaT** [comment]
Hello ,


I can not find any evidence ,

for me calling the number and having person answering is good proof , if
you do not accept it , you can proceed with closing.


Thank you


2015-08-05 13:31 GMT+02:00 Keghani Kouzoujian <notifications@github.com>:

> Hi again,
>
> Just to check in on this, will you provide evidence for the range you want
> supported? Otherwise we can't make the required change and will have to
> close this issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/libphonenumber/issues/754#issuecomment-127962901>
> .
>



-- 
Merci,
Cordialement.



----------------------------------------------

Alaa TOUNSADI
Ingénieur en Télécoms
GSM : 07 62 03 79 92

**Comment by keghani** [comment]
Hi AlaaT,

It's easy enough to verify this single number, but what about other similar numbers? Should we support all numbers of the form

    +213 23 xx xx xx or
    +213 23 2x xx xx or
    +213 23 29 xx xx or
    something else?

What's the prefix or the range that we should support? We need evidence or documentation for the whole range.

If you could help me formalize the range I'd be happy to support it.

**Comment by dlyongemallo** [comment]
We've found evidence for the +213 23 range. This will be added in the next version.

**Comment by NeverMindNever** [comment]
Oh that great thank you , 

I called the regulations authority but they did not gave me any detailed numbering plan.

Have a nice day.

**Comment by keghani** [comment]
This looks to be fixed: see +21323290096 at http://libphonenumber.appspot.com/
